src/tests/multihost/ad/test_dyndns.py

```diff
@@ -0,0 +1,308 @@
+        assert cmd_ptr_lookup.returncode != 0
+
+    def test_0003_verify_with_dyndns_ttl_functionality(self, multihost, adjoin, getptr):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: Verify with dyndns ttl functionality
+        :id: ffb8ee9b-2a82-4917-82b0-d87fd066a24c
+        :steps:
+          1. Join client to AD domain
+          2. Configure sssd.conf with dyndns_ttl = 9200
+          3. Perform DNS lookup for hostname
+          4. Perform DNS lookup for IP
+        :expectedresults:
+          1. Client joins domain
+          2. Should succeed
+          3. Hostname resolves with 9200 ttl
+          4. IP resolves with 9200 ttl
+        """
+        adjoin(membersw='adcli')
+        ttl = '9600'
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        domain = multihost.ad[0].domainname
+        hostname_cmd = multihost.client[0].run_command('hostname', raiseonerr=False)
+        hostname = hostname_cmd.stdout_text.rstrip()
+        (ip, network, ptr) = getptr
+        cmd_change_hostname = 'hostname ' + hostname
+        multihost.client[0].run_command(cmd_change_hostname)
+        domain_section = 'domain/{}'.format(domain)
+        sssd_params = {'dyndns_ttl': ttl}
+        client.sssd_conf(domain_section, sssd_params)
+        client.clear_sssd_cache()
+
+        time.sleep(30)
+        cmd_ptr_lookup = multihost.client[0].run_command(f"nslookup {ip}")
+        assert cmd_ptr_lookup.returncode == 0
+
+    def test_0004_check_dyndns_iface_with_existing_interfaces(
+            self, multihost, adjoin, extra_network, extra_interface, getptr):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: check dyndns iface with existing interface
+        :id: 166ee0a0-6f19-4be0-b5db-7373dde82862
+        :steps:
+          1. Join client to AD domain
+          2. Create dummy interface
+          3. Perform DNS lookup for hostname on dummy interface
+          4. Perform DNS lookup for IP on dummy interface
+        :expectedresults:
+          1. Client joins domain
+          2. Should succeed
+          3. Hostname resolves correctly on dummy interface
+          4. IP resolves correctly on dummy interface
+        """
+        (dummy_interface, dummy_ip) = extra_interface
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        domain = multihost.ad[0].domainname
+        hostname_cmd = multihost.client[0].run_command('hostname', raiseonerr=False)
+        hostname = hostname_cmd.stdout_text.rstrip()
+
+        adjoin(membersw='adcli')
+        domain_section = 'domain/{}'.format(domain)
+        sssd_params = {'dyndns_iface': dummy_interface}
+        client.sssd_conf(domain_section, sssd_params)
+        client.clear_sssd_cache()
+
+        multihost.client[0].run_command(f"nslookup {hostname}")
+        multihost.client[0].run_command(f"nslookup {dummy_ip}", raiseonerr=False)
+        cmd_lkup = multihost.client[0].run_command(f"nslookup {ip}", raiseonerr=False)
```

Comment:
I do not see definition of ip anywhere in the test so nslokup will fail and rerurncode is indeed !=0.